app/scripts/modules/ecs/src/serverGroup/configure/serverGroupConfiguration.service.ts

```diff
@@ -625,6 +627,21 @@ export class EcsServerGroupConfigurationService {
     return result;
   }
 
+  public checkDirtyCapacityProviders(command: IEcsServerGroupCommand): void {
+    if(command.capacityProviderStrategy && command.backingData.filtered.availableCapacityProviders){
```

Comment:
Shouldn't we also check this, if there are no available capacity providers? (if you delete all capacity providers associated to your account, you can still have previously selected an invalid one)

Reply:
Actually, `command.backingData.filtered.availableCapacityProviders` also holds true even if length of the availableCapacityProviders is zero which takes care of your concern. But if you want I can remove that condition to avoid the confusion. LMK your preference. @piradeepk

Reply:
Understood, my concern was if availableCapacityProviders somehow came back as null, this would not run, but the cp would be invalid.

Reply:
Than removing `command.backingData.filtered.availableCapacityProviders` will resolve the issue.